src/libs/actions/PolicyConnections.ts

```diff
@@ -0,0 +1,38 @@
+import Onyx from 'react-native-onyx';
+import type {OnyxUpdate} from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {OpenPolicyAccountingPageParams} from '@libs/API/parameters';
+import {READ_COMMANDS} from '@libs/API/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+function openPolicyAccountingPage(policyID: string) {
+    const hasConnectionsDataBeenFetchedKey = `${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${policyID}` as const;
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: hasConnectionsDataBeenFetchedKey,
+            value: false,
+        },
+    ];
+    const finallyData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: hasConnectionsDataBeenFetchedKey,
+            value: true,
+        },
+    ];
```

Comment:
@hayata-suenaga After playing with this, I think you missed the `failureData` setting `hasConnectionsDataBeenFetched` to `false` or something like that?

If you do:

1. Create a workspace
2. Enable `Accounting`, but don't got to the accounting settings yet
3. Disable your internet connection
4. Open the `Accounting` settings

I would have expected that an Offline blocker would appear since we haven't loaded the policy connections, but it doesn't because `withPolicyConnections` is set `true` even if `OPEN_POLICY_ACCOUNTING_PAGE` fails

Reply:
<img width="960" alt="image" src="https://github.com/Expensify/App/assets/87341702/9a0a600f-efc3-4d26-8d48-17f42c74f7c5">

<img width="1159" alt="image" src="https://github.com/Expensify/App/assets/87341702/cea8e4db-87cb-44ac-8a8f-14004d747359">

Reply:
[Here is the PR to fix the issue](https://github.com/Expensify/App/pull/40375). Thank you for letting me know of the issue 🙇